presto-main/src/main/java/com/facebook/presto/operator/HashSemiJoinOperator.java

```diff
@@ -136,18 +137,20 @@ public void addInput(Page page)
     {
         requireNonNull(page, "page is null");
         checkState(!finishing, "Operator is finishing");
+        ChannelSet channelSet = this.channelSet;
```

Comment:
Do we really need this line?

Reply:
It establishes `channelSet` as an effectively-final reference for use inside the subsequent loop body (one that is statically known to be non-null as well) whereas the instance field is not `final` and so the JIT may or may not leave in per-iteration null deoptimization traps.

Reply:
OK, that might worth a comment to this line....

Reply:
Comment added.